### PR TITLE
sys/trickle: migrate from xtimer to ZTIMER_MSEC

### PR DESCRIPTION
**Pull request by haukepetersen**
### Contribution description
This PR migrates `trickle` from xtimer to `ZTIMER_MSEC`. This allows for a) slightly simpler code, and b) for potential energy savings as `xtimer` can (at one point) be left out of a GNRC build.

The acutal changes are pretty straight forward, no surprises here :-) 

### Testing procedure
Running the `test/trickle` test should suffice.

### Issues/PRs references
none

### CODE REVIEW

**Comment by jue89** [comment]
All in all a valid change. But your PR shows up a problem, if just `ztimer_msec` is required. It tries to back its time by `ztimer_periph_timer` - but that module is missing. This must be fixed before merging this PR.

**Comment by haukepetersen** [comment]
Yap, but this is IMHO more a generic problem with the ztimer deps. I proposed a generic fix in #16334, lets see if it gets in :-)

**Comment by haukepetersen** [comment]
On top there is a problem in the `rpl` shell command, as it accesses internals of the trickle data structure directly, very good coding style :-)

But as it seems the calculation of the remaining trickle time is based on a xtimer function that does not exist for ztimer, so there is no easy drop-in replacement. Will need  to think of a viable approach to migrate that piece of code.

**Comment by haukepetersen** [comment]
After thinking about it, I will adapt this PR as well to be 'dual-timer', so to use ztimer only if its included in the build anyway and fallback to xtimer in any other case...

**Comment by kfessel** [comment]
#13667

**Comment by haukepetersen** [comment]
Adapted this PR to follow the same style as #16340 and #16339: `ztimer_msec` is now used only if available, whereas the default dependencies are not changed - despite including `xtimer` as dep only in the case `ztimer_msec` is not present in the build.

@jue89 This should also rule out the dependency problem we discussed.

**Comment by haukepetersen** [comment]
rebased

**Comment by haukepetersen** [comment]
rebased

**Comment by cgundogan** [comment]
The code changes look reasonable.  Just a quick question on how to test this .. I wanted to run the `trickle` test in `tests/` and modified the `Makefile` to include `USEMODULE += ztimer_msec`.  But I get an error from the linker:
```
RIOT/sys/ztimer/auto_init.c:224: undefined reference to `ztimer_periph_timer_init'
```

I was testing with `native`; is this expected because of missing adaptations?

**Comment by miri64** [comment]
> ```
> RIOT/sys/ztimer/auto_init.c:224: undefined reference to `ztimer_periph_timer_init'
> ```
> 
> 
>     
>   
> 
> I was testing with `native`; is this expected because of missing adaptations?

For `native` you need to manually include `ztimer_usec` at the moment.

**Comment by cgundogan** [comment]
> > ```
> > RIOT/sys/ztimer/auto_init.c:224: undefined reference to `ztimer_periph_timer_init'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I was testing with `native`; is this expected because of missing adaptations?
> 
> For `native` you need to also manually include `ztimer_usec` at the moment.

mhh, okay.  Works with the following diff:
```diff
diff --git a/tests/trickle/Makefile b/tests/trickle/Makefile
index 72efd754bc..5512f1cb0d 100644
--- a/tests/trickle/Makefile
+++ b/tests/trickle/Makefile
@@ -1,5 +1,7 @@
 include ../Makefile.tests_common
 
 USEMODULE += trickle
+USEMODULE += ztimer_usec
+USEMODULE += ztimer_msec
 
 include $(RIOTBASE)/Makefile.include
```

The print output of the test also looks correct.  I'd be fine with merging if there are no unresolved discussions (from above).

**Review by cgundogan — APPROVED** [review]
Reasonable code changes and the `tests/trickle` test shows expected results.

**Comment by miri64** [comment]
The Github action figured out just fine that the PR needs squashing. No need to set the label and possibly waste valuable CI time as such ;-).

**Comment by haukepetersen** [comment]
squashed.

**Comment by haukepetersen** [comment]
Some `gnrc_networking` example builds failed due to `sc_gnrc_rpl` using the timer member of the trickle timer directly, which lead to conflicts between ztimer and xtimer. I fixed this by pulling in the `sc_gnrc_rpl` fixes already included in #16339.

File: sys/shell/commands/sc_gnrc_rpl.c [hunk 1]
```diff
@@ -298,6 +299,8 @@ int _gnrc_rpl_dodag_show(void)
                 gnrc_rpl_instances[i].mop, gnrc_rpl_instances[i].of->ocp,
                 gnrc_rpl_instances[i].min_hop_rank_inc, gnrc_rpl_instances[i].max_rank_inc);
 
+#if !IS_USED(MODULE_XTIMER)
```
Comment (cgundogan) [inline sys/shell/commands/sc_gnrc_rpl.c:302]: Hmm, if there is no *better* way to access the current time-to-fire of the timer, then we should probably remove the TC print in both cases.  It was there for debugging purposes and is not very important to have that around.

Comment (haukepetersen) [reply sys/shell/commands/sc_gnrc_rpl.c:302]: No, to my knowledge there is not really a decent way with ztimer to get the TC. Removing all-together sounds good to me, will adapt.

**Comment by cgundogan** [comment]
The changes still look good after the amendments.

**Pull request merged**